src/main/java/com/amazon/aws/iot/greengrass/semver/Range.java

```diff
@@ -0,0 +1,486 @@
+                    ret = String.format(">=%s.%s.0 <%s.%s.0-0", M, m, M, Integer.parseInt(m) + 1);
+                } else {
+                    ret = String.format(">=%s.%s.0 <%s.0.0-0", M, m, Integer.parseInt(M) + 1);
+                }
+            } else if (pr != null) {
+                if ("0".equals(M)) {
+                    if ("0".equals(m)) {
+                        ret = String.format(">=%s.%s.%s-%s <%s.%s.%s-0", M, m, p, pr, M, m, Integer.parseInt(p) + 1);
+                    } else {
+                        ret = String.format(">=%s.%s.%s-%s <%s.%s.0-0", M, m, p, pr, M, Integer.parseInt(m) + 1);
+                    }
+                } else {
+                    ret = String.format(">=%s.%s.%s-%s <%s.0.0-0", M, m, p, pr, Integer.parseInt(M) + 1);
+                }
+            } else {
+                if ("0".equals(M)) {
+                    if ("0".equals(m)) {
+                        ret = String.format(">=%s.%s.%s <%s.%s.%s-0", M, m, p, M, m, Integer.parseInt(p) + 1);
+                    } else {
+                        ret = String.format(">=%s.%s.%s <%s.%s.0-0", M, m, p, M, Integer.parseInt(m) + 1);
+                    }
+                } else {
+                    ret = String.format(">=%s.%s.%s <%s.0.0-0", M, m, p, Integer.parseInt(M) + 1);
+                }
+            }
+
+            return ret;
+        }
+
+
+        // Because * is AND-ed with everything else in the comparator,
+        // and '' means "any version", just remove the *s entirely.
+        static String replaceStars(String comp) {
+            // Looseness is ignored here.  star is always as loose as it gets!
+            return SemVerRegexes.re.get(SemVerRegexes.t.get("STAR")).matcher(comp).replaceAll("");
+        }
+
+        static String replaceGTE0(String comp) {
+            return SemVerRegexes.re.get(SemVerRegexes.t.get("GTE0")).matcher(comp).replaceAll("");
+        }
+
+        // This function is passed to string.replace(re[t.HYPHENRANGE])
+        // M, m, patch, prerelease, build
+        // 1.2 - 3.4.5 => >=1.2.0 <=3.4.5
+        // 1.2.3 - 3.4 => >=1.2.0 <3.5.0-0 Any 3.4.x will do
+        // 1.2 - 3.4 => >=1.2.0 <3.5.0-0
+        private static String hyphenReplace(String $0, String from, String fM, String fm, String fp, String fpr,
+                                            String fb, String to, String tM, String tm, String tp, String tpr,
+                                            String tb) {
+            if (isX(fM)) {
+                from = "";
+            } else if (isX(fm)) {
+                from = String.format(">=%s.0.0", fM);
+            } else if (isX(fp)) {
+                from = String.format(">=%s.%s.0", fM, fm);
+            } else {
+                from = String.format(">=%s", from);
+            }
+
+            if (isX(tM)) {
+                to = "";
+            } else if (isX(tm)) {
+                to = String.format("<%s.0.0-0", Integer.parseInt(tM) + 1);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;+&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 434. Consider formatting this data with &#39;%d&#39; instead.